blueprints/deno-kv-litefs.html.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Deno KV with LiteFS Cloud
+layout: docs
+sitemap: false
+nav: firecracker
+author: jesse
+date: 2024-05-28
+---
+
+## Problem
+
+If you're porting an app from Deno Deploy to Fly.io, you may hit a bit of a snag: if it was built with [Deno KV](https://deno.com/blog/kv), a _traditionally_ FoundationDB backed K/V store made for Deno Deploy, it may seem like you're stuck because there's no obvious way to easily run Deno KV at scale on Fly.io. You _could_ use [denoland/denokv](https://github.com/denoland/denokv) on Fly.io, but due to an alignment of the stars there happens to be an easier way.
+
+Deno KV can actually be backed by an SQLite DB stored on disk in a cache folder, and you can actually specify the path of this DB with the parameter to `Deno.openKv( <here> )`.
+
+This means you can do something like this:
+
+```typescript
+const kv = await Deno.openKv("/any/path/i/want.db");
+```
+
+To get the SQLite-backed Deno KV implementation to work off an arbitrary SQLite DB.
+
+Now, Fly.io has some special features for SQLite users, namely [LiteFS Cloud](https://fly.io/docs/litefs/speedrun), a distributed file system that transparently replicates and backs up SQLite databases across all your instances. The magic here is that you can just treat it like a local on-disk SQLite database but behind the curtain it’s doing all the work to replicate your DB.
+
+## Setting up LiteFS
+
+Firstly, you need to add a `litefs.yml` file to your project, and make sure it gets included in the Dockerfile.
+
+Here’s an example:
+
+```yml
+fuse:
+  dir: "/litefs"
+
+data:
+  dir: "/var/lib/litefs"
+
+exit-on-error: false
+
+proxy:
+  addr: ":8080"
+  target: "localhost:8081"
+  db: "db"
+  passthrough:
+    - "*.ico"
+    - "*.png"
+
+exec:
+  - cmd: "deno run -A main.ts"
+
+lease:
+  type: "consul"
+  advertise-url: "http://${HOSTNAME}.vm.${FLY_APP_NAME}.internal:20202"
+  candidate: ${FLY_REGION == PRIMARY_REGION}
+  promote: true
+
+  consul:
+    url: "${FLY_CONSUL_URL}"
+    key: "litefs/${FLY_APP_NAME}"\
+```
+
+This is 99% the [sample config](https://github.com/superfly/litefs-example/blob/main/fly-io-config/etc/litefs.yml), only changing the exec command to a Deno one.
+
+<section class="callout"> It’s not immediately clear based on the config, but you need to make your app listen on 8081, not 8080. Part of LiteFS acts as a proxy, so you need the fly-proxy to send requests to LiteFS which then forwards them on to your app.
+
+If you’re not able to change your application’s port, make sure `target` is set up correctly then change `addr` and `fly.toml` to something other than `:8080`. </section>
```

Comment:
Suggestion to make it an "important" note. Also, maybe explicitly say that litefs uses port 8080 by default?
```suggestion
<div class="important icon"> 
It’s not immediately clear based on the config, but you need to make your app listen on 8081, not 8080. Part of LiteFS acts as a proxy, and listens on port 8080 by default ?? so you need the fly-proxy to send requests to LiteFS which then forwards them on to your app.
If you’re not able to change your application’s port, make sure `target` is set up correctly then change `addr` and `fly.toml` to something other than `:8080`. 
</div>
```